code/modules/dwarven/dwarven_structures.dm

```diff
@@ -0,0 +1,145 @@
+/obj/structure/destructible/dwarven
+	density = TRUE
+	anchored = TRUE
+	icon = 'icons/obj/dwarven.dmi'
+	light_power = 1
+	var/cooldowntime = 0
+	//break_sound = 'sound/hallucinations/veryfar_noise.ogg' //to do: find a suitable noise
+	//debris = list(/obj/item/stack/sheet/runed_metal = 1) //to do : make a dwarven metal datum
+
+/obj/structure/destructible/dwarven/New()
+	START_PROCESSING(SSfastprocess, src)
+	..()
+
+/obj/structure/destructible/dwarven/Destroy()
+	STOP_PROCESSING(SSfastprocess, src)
+	return ..()
+
+
+/obj/structure/destructible/dwarven/dwarven_sarcophagus
+	name = "Ancient Sarcophagus"
+	icon_state = "sarcophagus_open"
+
+	var/recharge = TRUE
+	var/recharge_points = 0
+	var/recharge_points_max = 500
+
+/obj/structure/destructible/dwarven/dwarven_sarcophagus/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>It requires [recharge_points_max-recharge_points] points to reactivate </span>"
+
+/obj/structure/destructible/dwarven/dwarven_sarcophagus/attackby(obj/item/stack/ore/I, mob/living/user, params)
+	recharge_points += I.amount * I.points
+	qdel(I)
+	check_requirements()
+	return
+
+/obj/structure/destructible/dwarven/dwarven_sarcophagus/proc/check_requirements()
+	if(recharge_points_max <= recharge_points)
+		for(var/mob/M in viewers(src,5))
+			to_chat(M, "<span class='notice'>The sarcophagus reignites with ancient fire, ready to birth another dwarf!</span>")
+			new  /obj/effect/mob_spawn/human/dwarven_sarcophagus(get_turf(src))
+			qdel(src)
+	else
+		for(var/mob/M in viewers(src,5))
+			to_chat(M, "<span class='notice'>The sarcophagus is set ablaze for a second, but ancient powers die down. It requires more minerals!</span>")
+
+/obj/structure/destructible/dwarven/lava_forge
+	name = "Ancient Forge"
+	icon_state = "dwarven_forge"
+
+
+/obj/structure/destructible/dwarven/lava_forge/attackby(obj/item/stack/ore/I, mob/living/user, params)
+	var/refined = I.refined_type
+	if(isnull(refined))
+		return
+	else
+		var/probability = (rand(100,200)/100)
```

Comment:
You're not gonna make a PR thats literaly dwarf fortress and not add a smithing skill, are you. that'd be very unepic :^)